### PR TITLE
CDAP-9316: Use LoggerName if tags cannot be added using caller Data

### PR DESCRIPTION
**Pull request by deepakwadhwani**

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java [hunk 1]
```diff
@@ -77,33 +77,47 @@ private void addExtraTags(ILoggingEvent event) {
...
-          }
-          break;
-        }
-      }
+      tags = addTagsForClass(callerClass, callerClasses);
 
       loggerExtraTags.put(callerClass, tags);
     }
 
+    // If no tags were added using calledData, try using LoggerName
```
Comment (ecfm) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java]: `callerData` misspelled

**Comment by chtyim** [comment]
When will this happen?

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java [hunk 2]
```diff
@@ -77,33 +77,47 @@ private void addExtraTags(ILoggingEvent event) {
...
       loggerExtraTags.put(callerClass, tags);
     }
 
+    // If no tags were added using callerData, try using LoggerName
+    if (tags.isEmpty()) {
+      tags = loggerExtraTags.getIfPresent(event.getLoggerName());
+      if (tags == null) {
+        tags = addTagsForClass(event.getLoggerName(), callerClasses);
+      }
+      loggerExtraTags.put(event.getLoggerName(), tags);
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java]: No need to put if it is empty

**Comment by deepakwadhwani** [comment]
@chtyim Like discussed in the past, this happens for logs from within the script of the Javascript Transform. Additional details are in the bug showing the callerData and the execution stack for those logs.

**Comment by ecfm** [comment]
LGTM. Tested on cluster with JS and TopN plugin successfully.

**Comment by deepakwadhwani** [comment]
Running build: https://builds.cask.co/browse/CDAP-RUT890-1

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java (hunk 2 above)
Comment (chtyim) [reply cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java]: hum... forget it. should be ok.

**Comment by chtyim** [comment]
LGTM

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java (hunk 2 above)
Comment (deepakwadhwani) [reply cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogAppender.java]: yes, it saves us from trying over and over for some callerData/LoggerName that does not match with any frame on the execution stack. Although I should move the put inside the null check. Will update the diffs.

**Pull request merged**